src/mx/SourceInterface.java

```diff
@@ -0,0 +1,7 @@
+package mx;
+
+import java.util.ArrayList;
+
+public interface SourceInterface {
+    ArrayList<Domain> getDomainList() throws SourceNotFoundException;
```

Comment:
Return abstract type, that suits your requirements. `List` interface maintains order - is it required here? Maybe `Collection` would be sufficient?
There are four types of collection semantics in general: collection (just a grouping actually - no additional constraints), set (guarantees uniqueness in mathematical sense), list (maintains order) and map (denotes association). Always try to think which of those you need when creating APIs.